pkg/executor/newdeploy/newdeploy.go

```diff
@@ -428,7 +429,10 @@ func (deploy *NewDeploy) waitForDeploy(depl *v1beta1.Deployment, replicas int32)
 		}
 		time.Sleep(time.Second)
 	}
-	return nil, errors.New("failed to create deployment within timeout window")
+
+	// this error appears in the executor pod logs
+	timeoutError := fmt.Sprintf("failed to create deployment within timeout window of %d seconds", waitTimeOut)
```

Comment:
You can use `fmt.Errorf` here.

Reply:
Makes sense. I have made the change. Thank you :)